File: src/main/java/com/imp/checklist/ChecklistTemplate.java
package com.imp.checklist;

import com.imp.checklist.question.Question;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;

@Getter
@Setter
@Document("checklistTemplate")
public class ChecklistTemplate implements Checklist{
    @Id
    private java.lang.String id;
    private ArrayList<String> artifactTypes;
    private ArrayList<Question> questions;
}

File: src/main/kotlin/com/julia/imp/checklist/ChecklistTemplate.kt
package com.julia.imp.checklist

import kotlinx.serialization.Contextual
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import org.bson.types.ObjectId

@Serializable
data class ChecklistTemplate(
    @Contextual
    @SerialName("_id")
    val id: ObjectId,
    val artifactTypes: List<String>,
    val questions: List<String>
) : Checklist()
